pandas/core/base.py

```diff
@@ -219,6 +219,25 @@ def _obj_with_exclusions(self):
         else:
             return self.obj
 
+    @final
+    @cache_readonly
+    def _obj_numeric_only_with_exclusions(self):
```

Comment:
I think the performance of this change needs to be closely studied to see if it is really worth the complexity (and extra memory usage). In any case, I do not think it should be introduced here, but rather in its own PR. Can you follow the existing pattern of subsetting the data directly in the `agg` methods instead.

Reply:
related to comment I put below:

https://github.com/pandas-dev/pandas/pull/58132#discussion_r1569203503